viewer/packages/cad-geometry-loaders/src/sector/culling/ByVisibilityGpuSectorCuller.ts

```diff
@@ -161,32 +70,18 @@ export class ByVisibilityGpuSectorCuller implements SectorCuller {
   determineSectors(input: DetermineSectorsInput): { wantedSectors: WantedSector[]; spentBudget: SectorLoadingSpent } {
     const takenSectors = this.update(input.camera, input.cadModelsMetadata, input.clippingPlanes, input.budget);
     const wanted = takenSectors.collectWantedSectors();
-    const nonDiscarded = wanted.filter(x => x.levelOfDetail !== LevelOfDetail.Discarded);
-
-    const totalSectorCount = input.cadModelsMetadata.reduce((sum, x) => sum + x.scene.sectorCount, 0);
-    const takenSectorCount = nonDiscarded.length;
-    const takenSimpleCount = nonDiscarded.filter(x => x.levelOfDetail === LevelOfDetail.Simple).length;
-    const forcedDetailedSectorCount = nonDiscarded.filter(x => !Number.isFinite(x.priority)).length;
-    const accumulatedPriority = nonDiscarded
-      .filter(x => Number.isFinite(x.priority) && x.priority > 0)
-      .reduce((sum, x) => sum + x.priority, 0);
-    const takenDetailedPercent = ((100.0 * (takenSectorCount - takenSimpleCount)) / totalSectorCount).toPrecision(3);
-    const takenPercent = ((100.0 * takenSectorCount) / totalSectorCount).toPrecision(3);
+    const spentBudget = takenSectors.computeSpentBudget();
+
+    const takenDetailedPercent = (
+      (100.0 * (spentBudget.loadedSectorCount - spentBudget.loadedSectorCount)) /
```

Comment:
`spentBudget.loadedSectorCount - spentBudget.loadedSectorCount` is always 0. Not entirely sure what it should be instead here, though :thinking:

Reply:
lol, @larsmoa any input here?